fem/fe.cpp

```diff
@@ -12627,6 +12637,2047 @@ void ND_SegmentElement::CalcVShape(const IntegrationPoint &ip,
    obasis1d.Eval(ip.x, vshape);
 }
 
+ND_R1D_PointElement::ND_R1D_PointElement(int p)
+   : VectorFiniteElement(1, 2, 0, Geometry::POINT, 2, p,
+                         H_CURL, FunctionSpace::Pk)
+{
+   // VectorFiniteElement::SetDerivMembers doesn't support 0D H_CURL elements
+   // so we mimic a 1D element and then correct the dimension here.
+   dim = 0;
+}
+
+void ND_R1D_PointElement::CalcVShape(const IntegrationPoint &ip,
+                                     DenseMatrix &shape) const
+{
+   shape(0,0) = 1.0;
+   shape(0,1) = 0.0;
+
+   shape(1,0) = 0.0;
+   shape(1,1) = 1.0;
+}
```

Comment:
According to the documentation, this is an arbitrary order element that is constant in the second and third dimension. Why is just a constant 2D basis of dimension 2?

Reply:
It looks weird, doesn't it? This finite element is analogous to the `PointFiniteElement` which appears alongside the `H1_FiniteElements`. It's "arbitrary order" in the sense that it is compatible with 1D Nedelec finite elements of any order. In the constructor I pass an integer `p` even though this is not really relevant which is the "order" used in the `ND_R1D_FECollection`. Following the discussion in PR #2163 I'm really not sure what to do with these integers.

Reply:
This is the first time I've looked at finite elements at a point, and it doesn't make sense to me yet. I see that `H1_FECollection` has `PointFiniteElement`, and it seems to be just 0-order (constant). How is it actually used, and why? Does it just get used for evaluating a point value of a `GridFunction` or something, so it only needs to be a constant that gets multiplied by a value given by a coefficient?

Reply:
The only uses I can think of are as boundary elements. For example, when preparing a Neumann BC for a PDE on a 1D domain. This involves an integral over the boundary which in this case is just evaluating the BC function and the basis function at a single point. Rather than treating 0D boundaries as special cases we have 0D basis functions which can be plugged into the usual machinery and produce the proper results.

Reply:
Then maybe the source of confusion is the comment in the hpp file that the elements are "arbitrary order." The documentation for `PointFiniteElement` does not say anything about the order. Would it be more accurate to say it is 0-order and works for 1D boundary evaluations with arbitrary order 1D elements? Or is it better just not to mention order, since it can be confusing without a complete explanation of how these elements are used?

Reply:
Agreed. I rewrote the comment to remove "order" and hopefully summarized its purpose a little better.